### DISCUSSION TITLE
Partial override for `MapGenerationParameters` impossible?

### DISCUSSION THREAD

**Discussion by arousle** [FakeFishGames/Barotrauma#11734] [Code questions and modding support]
I'm new to modding Barotrauma, and for my first mod I wanted to do a single override to change the Jovian Radiation [`RadiationStep` to 20.0 (from its default value of 100.0)](https://github.com/Regalis11/Barotrauma/blob/6cc100d98cc99e758f4ea0750528a2aa68429c0a/Barotrauma/BarotraumaShared/SharedSource/Map/Map/RadiationParams.cs#L15).

After reading the [modding guide on overrides](https://regalis11.github.io/BaroModDoc/Intro/Overrides.html), I thought I could simply do this:
```xml
<!-- filelist.xml -->
<contentpackage
    name="Slowed (20%) Jovian Radiation"
    modversion="1.0.0"
    gameversion="1.0.8.0"
    corepackage="false"
>
    <MapGenerationParameters file="%ModDir%/mapGenerationParameters.xml" />
</contentpackage>
```
```xml
<!-- mapGenerationParameters.xml -->
<Override>
    <MapGenerationParameters>
        <RadiationParams
            radiationstep="20.0"
        />
    </MapGenerationParameters>
</Override>
```

But after (a frustrating amount of) trial-and-error, it seems that this sort of partial override isn't actually possible. It has been my experience that any _partial_ override of the `MapGenerationParameters` either crashes the game upon map generation or causes other unexpected behavior to occur. Further evidence to corroborate this is that all examples of mods on GitHub which override the `MapGenerationParameters` do so by fully specifying _all_ of its parameters (e.g. [(1)](https://github.com/panhao4812/GameTools/blob/19855f50a984f15c5c92bde78780e1c60907b67f/BaroAngEditor/BaroAngEditor/mods/MapGenerationParameters_Enemies.xml), [(2)](https://github.com/Udrakan/baroita/blob/0c4def2b4b59f9ee7509ee36cfb3b7eab4dbd0b1/Content/Map/ita_mapgenerationparameters.xml)).

I understand this means that only one mod's changes to `MapGenerationParameters` can ever be active, which doesn't seem very modular. 
Additionally, each mod which does not want to change even a single `MapGenerationParameters` from the game's defaults would have to be updated if _any_ of the defaults change, which doesn't seem very future-proof.

Is this inability to provide partial overrides to `MapGenerationParameters` working as intended? Am I missing something?

**Comment by BiffW** [comment]
Actually, any form of partial override is not supported officially.
There's an unoffical branch that provided partial override for items and characters for now.
You can have access to it by installing this: https://steamcommunity.com/sharedfiles/filedetails/?id=2856758496

**Reply by arousle** [reply]
Have the devs given any word on whether they'll address this issue? Do they even _see_ it as an issue?

Thanks for the info on the mod. Shame that pull request looks like a mess, to be honest, and I doubt it'll ever be merged.
